flink-cdc-connect/flink-cdc-source-connectors/flink-cdc-base/src/main/java/org/apache/flink/cdc/connectors/base/source/reader/IncrementalSourceReader.java

```diff
@@ -541,4 +542,11 @@ private void logCurrentStreamOffsets(List<SourceSplitBase> splits, long checkpoi
             LOG.info("Stream split offset on checkpoint {}: {}", checkpointId, offset);
         }
     }
+
+    @Override
+    public void close() throws Exception {
+        // clear jdbc connection pools
+        JdbcConnectionPools.close();
```

Comment:
Some abstraction is needed here because the sources we support are not limited to just the jdbc datasource. Perhaps we should delegate the implementation of closing the connection pool to DataSourceDialect.

Reply:
Thanks, it looks better